### PR TITLE
OCPBUGS-19695: gather APIServer.config.openshift.io resource

### PR DESCRIPTION
**Pull request by ncaak**
This PR implements a new gatherer that collects **apiserver.config.openshift.io resource**.

## Categories

- [ ] Bugfix
- [ ] Data Enhancement
- [X] Feature
- [ ] Backporting
- [ ] Others (CI, Infrastructure, Documentation)

## Sample Archive

- `docs/insights-archive-sample/config/apiserver.json`

## Documentation

- `docs/gathered-data.md`

## Unit Tests

- `pkg/gatherers/clusterconfig/gather_openshift_config_apiserver_test.go`

## Privacy

Yes. There are no sensitive data in the newly collected information.

## Changelog

No

## Breaking Changes

No

## References

https://issues.redhat.com/browse/CCXDEV-11644
https://issues.redhat.com/browse/OCPBUGS-19695

### CODE REVIEW

File: pkg/gatherers/clusterconfig/gather_openshift_config_apiserver.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+func (g *Gatherer) GatherOpenshiftConfigAPIServer(ctx context.Context) ([]record.Record, []error) {
+	configClient, err := configv1client.NewForConfig(g.gatherKubeConfig)
+	if err != nil {
+		return nil, []error{err}
+	}
+
+	return configAPIServer{}.gather(ctx, configClient.APIServers())
+}
+
+type configAPIServer struct{}
```
Comment (tremes) [inline pkg/gatherers/clusterconfig/gather_openshift_config_apiserver.go:43]: Is this type needed?

Comment (ncaak) [reply pkg/gatherers/clusterconfig/gather_openshift_config_apiserver.go:43]: for the logic? no. But I think it brings some readability and avoids charging clusterconfig package with more unexported functions.

Empty structs are practically free and this way of presenting the code encloses better the scope and gives you more flexibility on function naming.

But, I don't have a strong opinion on this, if you prefer I can change it to same name as the exported function.

**Comment by JoaoFula** [comment]
cluster setup issue.

**Comment by tremes** [comment]
Reviewed & verified. Thanks.
/lgtm

**Comment by JoaoFula** [comment]
Verification is on my side RN. Will provide qe-approved label only after verification due to broken qe-approved/OCPBUGS process (see [OCPCRT-268](https://issues.redhat.com/browse/OCPCRT-268))

**Pull request merged**